spring-integration-mail/src/main/java/org/springframework/integration/mail/AbstractMailReceiver.java

```diff
@@ -553,9 +595,25 @@ Store getStore() {
 
 		private final MimeMessage source;
 
+		private final Object content;
+
 		private IntegrationMimeMessage(MimeMessage source) throws MessagingException {
 			super(source);
 			this.source = source;
+			if (AbstractMailReceiver.this.simpleContent) {
+				this.content = null;
+			}
+			else {
+				Object complexContent;
+				try {
+					complexContent = source.getContent();
+				}
+				catch (IOException e) {
+					complexContent = "Unable to extract content; see logs: " + e.getMessage();
+					AbstractMailReceiver.this.logger.error("Failed to extract content from " + source, e);
+				}
+				this.content = complexContent;
```

Comment:
Yeah. OK. I see the code in the `MimeMessage(MimeMessage source)`, where we have this code in the `parse()` method:
```
try {
	content = ASCIIUtility.getBytes(is);
} catch (IOException ioex) {
	throw new MessagingException("IOException", ioex);
}
```
So, why don't follow with the same approach and don't log exception but re-throw as similar `MessagingException`?